File: level1/달리기 경주/Solution.java
import java.util.*;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        Map<String, Integer> rank = new HashMap<>();

        // 선수별 현재 위치 저장
        for (int i = 0; i < players.length; i++) {
            rank.put(players[i], i);
        }

        for (String calling : callings) {

            // 추월한 선수의 현재 위치
            int currentIndex = rank.get(calling);

            // 바로 앞 선수
            String frontPlayer = players[currentIndex - 1];

            // 배열에서 두 선수 위치 교환
            players[currentIndex - 1] = calling;
            players[currentIndex] = frontPlayer;

            // HashMap의 위치도 수정
            rank.put(calling, currentIndex - 1);
            rank.put(frontPlayer, currentIndex);
        }

        return players;
    }
}

File: level1/달리기 경주/Solution.py
def solution(players, callings):
    # 선수 이름별 현재 위치
    positions = {name: i for i, name in enumerate(players)}

    for name in callings:
        current = positions[name]
        ahead_name = players[current - 1]

        # 두 선수의 순서와 위치 갱신
        players[current - 1], players[current] = players[current], players[current - 1]
        positions[name] = current - 1
        positions[ahead_name] = current

    return players